### ISSUE TITLE
Hyphen in station name will cause error message

### ISSUE THREAD

**Issue by swilson86** [node-red/node-red-nodes#782]
If a hyphen appears in a station name ie: Cistern-Pump on a local LAN, ping 3.0 will generate an error message "ping: Cistern-Pump: Name or service not known". This is on a raspberry pi running Buster. Pinging any station without a hyphen works. I also ran the suggested 2 lines of code in the documentation on the pi with same results afterwards.

**Comment by knolleary** [comment]
Hi @swilson86 

Please try to use the issue template we provide as it helps to ensure all the right information is provided.

As this repository contains dozens of nodes, can you confirm which node you are reporting an issue with?

**Comment by dceejay** [comment]
@meeki007 I'm guessing this is the ping node ? Any chance you can take a look ?

**Comment by meeki007** [comment]
Will take a look into issue today. If not found/figured out in an hour I can give it tons of attention this Saturday in 36 hours

**Comment by meeki007** [comment]
### Which node are you reporting an issue on?
node-red-node-ping - 88-ping https://github.com/node-red/node-red-nodes/tree/master/io/ping

### What are the steps to reproduce?
Edit /etc/hosts file to translate hostname of Cistern-Pump to a known good ip

![https://i.ibb.co/mbnkLPR/pingipadd.jpg](https://i.ibb.co/mbnkLPR/pingipadd.jpg)

$ sudo nano /etc/hosts

![https://i.ibb.co/z2CfH6w/nanohosts.jpg](https://i.ibb.co/z2CfH6w/nanohosts.jpg)

Use ping node to ping hostname

![https://i.ibb.co/jHywRXG/node-redexamp.jpg](https://i.ibb.co/jHywRXG/node-redexamp.jpg)


### What happens?

It works just fine

### What do you expect to happen?

User stated that it does not resolve ---> "ping: Cistern-Pump: Name or service not known"

### Please tell us about your environment:

- [X] Node-RED version: 1.2.9
- [X] node.js version: v12.21.0
- [X] npm version: 6.14.11
- [X] Platform/OS: Linux - ubuntu - 20.04
- [X] Browser: FireFox 77.0.1


Request that user @swilson86 fills out proper issue template with focus on:
Platform/OS - there are lots of flavors of OS using buster. Is it the raspbian version/DietPI version/etc?
Link to documentation user states he used - "I also ran the suggested 2 lines of code in the documentation on the pi with same results afterwards"

Note to @swilson86 :
As you can see there are many of us who would love to help fellow users; please take the time to give us detailed steps needed to arrive at your issue. I have a extra raspberry pi(s) 2,3,4 just laying around. When having a beer this weekend I'd love to spin up a pi matching your config. Why? Its fun and I spent a good amount of time a few weeks ago pushing a update to the node-red-node-ping Node and its code is still fresh in my head.

**Comment by dceejay** [comment]
Thanks for taking a look - over to @swilson86  to respond.

**Comment by swilson86** [comment]
I apologize for not entering info in the correct format. I'm still isolating this apparent bug and will submit the results (properly) when I learn more about how to consistently reproduce it. It is a bug on the Ping version 3.0 node. I've got several Pi Raspbian OS computers and Windows 10. More info shortly. Thanks!!

**Comment by swilson86** [comment]
### Which node are you reporting an issue on?
node-red-node-ping version 0.3.0

### What are the steps to reproduce?
Setup a ping to a station on your LAN that does not exist and has never existed.... ie: Cistern-Pump or RainW etc. Any name will do. Just put in the name itself in the Target (ie: No http://RainW just enter RainW). Select Protocol Automatic, Mode Timed, and ping perhaps every 60s. I earlier suspected it was a hyphen causing the problem but that is not true as RainW will also fail.

### What happens?
In node-red debug window you will get the 2 red bars and the error "ping: RainW: Name or service not known" Placing a debug node just after the Ping node reveals that Ping does in fact return the false payload. Here is what you will see:

4/3/2021, 2:53:36 PMnode: Ping RainW
msg : string[39]
"ping: RainW: Name or service not known↵"
4/3/2021, 2:53:36 PMnode: Debug Ping RainW
RainW : msg : Object
object
payload: false
topic: "RainW"
_msgid: "a947235e.639e"

In the above example the station RainW has never logged into the LAN thus it is not in the routers routing table nor is it obviously reachable. On MS Windows it works OK without the error, but on Raspbian Buster it generates the error.

### What do you expect to happen?
Ping should not return the two red error bars and the "ping: RainW: Name or service not known↵" message. Ping exists for this purpose... to find out if a station is responding or not.

### Please tell us about your environment:

- [ ] Node-RED version: V1.2.9 running dashboard 2.28.2
- [ ] node.js version: v12.22.0
- [ ] npm version: 6.14.11
- [ ] Platform/OS: Raspian Buster on Pi 3B and Pi 4b (Linux 5.10.17-v7+ arm LE)
- [ ] Browser: Chromium Version 88.0.4324.187 (Official Build) Built on Raspbian , running on Raspbian 10 (32-bit)

**Comment by meeki007** [comment]
This is the correct behaviour in linux. RainW is unknown.
If I just ping RainW from my command line as its not been added to my network I get:
```
ping: RainW: Name or service not known
```
![https://i.ibb.co/jV9h85x/pingcmd.jpg](https://i.ibb.co/jV9h85x/pingcmd.jpg)

I get the same in node-red
![https://i.ibb.co/K0fspxP/ping001.jpg](https://i.ibb.co/K0fspxP/ping001.jpg)

Ran this on a raspberry pi running raspbian
### Please tell us about your environment:
- [X] Node-RED version: 1.2.9
- [X] node.js version: v12.21.0
- [X] npm version: 6.14.11
- [X] Platform/OS: 5.4.83-v7l armv7l GNU/Linux

I don't understand the issue. What does Windows output instead? I'm not a windows user or have a copy of windows.
Any one else here that is a Windows user understand what I'm missing?
This all looks like normal behaviour to me.

**Comment by swilson86** [comment]
Hi meeki007.  In node-red the two red bars on both sides of a message indicate some type of error. If it were changed to yellow which indicates a warning that would be better. But in the case of ping... if a pinged station does not exist it is logical that no error or warning should appear... the result should be just "false". Ping should not have anything to do with "this station existed but it does not now, this station never existed, etc." The answer to a Ping is the time it takes to respond or a "false". It drives me crazy looking a row of red bars that look like errors when I am visually scanning a debug output, only to find it is a failed Ping.

**Comment by dceejay** [comment]
Yes - maybe we should make this consistent - so node output is always a real number or "false" for any non-reply. And then maybe make the status under the node better reflect any extra info - but not raise any real error - as I agree that not being able to resolve a host is not an error from pings point of view.

**Comment by swilson86** [comment]
I almost agree with dceejay. The code to place extra info under the node should perhaps NOT be included in the next version. It is not expected nor needed. The standard should be "A node should not have any excess code and unexpected behavior" or it bloats the code. A new and separate node COULD be created that might be named PingDetails that returns the excess info for those that need it.

But in any case, we have identified a bug that exists in Linux/Raspian based systems on Ping version 0.3.0

**Comment by meeki007** [comment]
Sorry for the late reply. I was out of cell service range up in north
Wisconsin.

Will be back in Mishigan friday and take a look at implementing fixes per
dcjay's recommendations Saturday.

Open source is free but waits on us to get free time to contribute.
